### PR TITLE
Remove globals

### PR DESCRIPTION
**Pull request by jnoortheen**
# Todo

- [x] replace usage of builtins.__xonsh__ with module level singleton. while users will still have access to __xonsh__. 

# Objective 

- [x] Doing this, things will be more explicit and tests can run as single suit

# Follow up things to do
- [ ] `builtins.__xonsh__` should be deprecated while `__xonsh__` will be available to the session scope. 

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by daniel-shimon** [comment]
This is a great idea (which I actually also thought about doing this week 😅)!
Looking forward for it!

**Comment by jnoortheen** [comment]
Hi @daniel-shimon , most of the things are done. Now the failing tests number is around 23 when run with `pytest` in single invocation (atleast on my machine). Please help with the failing completers test. Also I have refactored the commits. You could start checking each commits in order.

**Comment by daniel-shimon** [comment]
The issue is that the `xonsh_builtins` fixture is now leaking.
This is because `XSH.load` is changing state (e.g. `env`) and that change isn't being reversed after the test.
Because `XSH` is now a singleton, I think its `__init__` needs to create all the attributes that are currently defined in `load`.
And then in `xonsh_builtins` just use `monkeypatch.setattr` instead of calling `load`, which will ensure these values will reset after the test

**Comment by jnoortheen** [comment]
> The issue is that the `xonsh_builtins` fixture is now leaking.
> This is because `XSH.load` is changing state (e.g. `env`) and that change isn't being reversed after the test.
> Because `XSH` is now a singleton, I think its `__init__` needs to create all the attributes that are currently defined in `load`.
> And then in `xonsh_builtins` just use `monkeypatch.setattr` instead of calling `load`, which will ensure these values will reset after the test

I don't think all the variables are available during __init__ . since fixture calls load each time, the attributes are reset. 

This two tests are using grep and dir.exe from system. also the bash_complete is not mocked. That is fine since this can work as integration test.

**Comment by daniel-shimon** [comment]
Yeah the idea is they use the normal PATH
so they don't expect XSH to be mocked

**Comment by jnoortheen** [comment]
Hi, @daniel-shimon I decided to deprecate the builtins usage in a follow-up PR. Do you want to take it?

**Comment by daniel-shimon** [comment]
Can you add the tests dir to `black` in `run-tests.xsh`?

**Comment by jnoortheen** [comment]
> Can you add the tests dir to `black` in `run-tests.xsh`?

it is added to the qa run https://github.com/xonsh/xonsh/pull/4280/commits/d1521f0bbef9ce71cde168c8b3689a49572c6a15

**Review by daniel-shimon — COMMENTED** [review]
Looks good!

File: tests/conftest.py [hunk 1]
```diff
@@ -104,63 +67,63 @@ def xonsh_events():
 
 
 @pytest.fixture
-def xonsh_builtins(monkeypatch, xonsh_events):
+def xonsh_builtins(monkeypatch, xonsh_events) -> XonshSession:
```
[the remark this answers is not in the document]
Comment (jnoortheen) [reply tests/conftest.py]: updated

File: tests/test_xonsh.xsh [hunk 2]
```diff
@@ -13,12 +10,13 @@ def test_envionment():
 
 
 def test_xonsh_party():
-  orig = builtins.__xonsh__.env.get('XONSH_INTERACTIVE')
-  builtins.__xonsh__.env['XONSH_INTERACTIVE'] = False
+  from xonsh.built_ins import XSH
+  orig = XSH.env.get('XONSH_INTERACTIVE')
+  XSH.env['XONSH_INTERACTIVE'] = False
```
[the remark this answers is not in the document]
Comment (jnoortheen) [reply tests/test_xonsh.xsh:15]: I already tried that. The pytest fixtures are not working with .xsh tests. So I manually unset it later.

**Review by daniel-shimon — APPROVED** [review]

**Comment by daniel-shimon** [comment]
@jnoortheen thanks!

**Pull request merged**